config/config.go

```diff
@@ -85,7 +85,7 @@ func Load(s string, expandExternalLabels bool, logger log.Logger) (*Config, erro
 		return nil, err
 	}
 
-	if !expandExternalLabels {
+	if cfg.GlobalConfig.ExternalLabels.Len() == 0 {
 		return cfg, nil
 	}
```

Comment:
I think we want to drop this conditional completely.

Reply:
But although external labels have been enabled by default, can still be left empty in the configuration file, and I'm worried that removing this conditional completely will cause some tests to failed.

Reply:
[Below in line 93](https://github.com/prometheus/prometheus/pull/14657/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R93) this code iterates over the configured external labels. So the check `cfg.GlobalConfig.ExternalLabels.Len() == 0` is redundant.